lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -101,7 +101,7 @@ SILPassPipelinePlan::getDiagnosticPassPipeline(SILOptions Options) {
   // disabled.
   if (Options.DebugSerialization) {
     addMandatoryDebugSerialization(P);
-    return P;
+    return const_cast<const SILPassPipelinePlan&>(P);
```

Comment:
What is this cast for? This probably defeats NRVO. Looking at the code here, I suspect the deleted move constructor was intended to prevent people from writing pessimizing `std::move`s, but Clang has a warning for that now. AFAICT it should be safe to just remove the `SILPassPipelinePlan(SILPassPipelinePlan &&) = delete;` declaration.